### DISCUSSION TITLE
Extracting nodes of ways from pbf file

### DISCUSSION THREAD

**Discussion by riprobins** [ropensci/osmextract#293] [Q&A]
I am running OSRM locally to extract the nodes that a trip traverses through.  I need to match this back to the way_ids from OSM.

I'm extracting the way_ids from the same pbf file I'm using in OSRM, and I want to extract the node ids from the PBF file, because those match the node ids returned from OSRM.

I can't see a way to use osmextract on the PBF file to get a list of way_id, node_id, longitude, latitude .

I would love some help please

**Comment by Robinlovelace** [comment]
@grobins do you have a reproducible example to share? That could help provide an answer.

**Comment by agila5** [comment]
Hi @grobins and thank you very much for your Q. Unfortunately, I think that current approach cannot recover the node_ids from a route. Anyway, as @Robinlovelace already mentioned, if somehow you can provide an example of the desired goal, maybe I can explore alternative tools (e.g. `osmdata`) and/or ask this question to the OSM/GDAL communities for feedbacks.
